### DISCUSSION TITLE
Graph download issues in media

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#100] [Q&A] [answered]
I changed my storage location from "data" to "media" and have noticed something odd when downloading, particularly new Graphhopper files.  This is a Samsung S20 running Android 13.

The download system on Graphs/Maps/POIs seems not to initiate a lot of the time despite a good WiFi connection and the URL being OK via the phone's browser.  When Cruiser starts later - it often launches several downloads as it attempts to play catch-up.  Maybe some kind of "cache" within the Android OS?

Interestingly, some of the Graphhopper files downloaded OK but the folder didn't name correctly - it was missing the "(Mapsforge)" text, so ended up with "\graphs\france" rather than "\graphs\france (Mapsforge)".  Renaming the folder manually within the file-system got it working.

Some patience and a bit of under-the-hood know-how helped me work around this, but I thought it worth posting about!

**Comment by devemux86** [comment, marked as answer]
@MotoUKRider thanks for the report!

Downloads are done by Android, they are always downloaded in the internal storage and data folder.
Then according to our selection in Cruiser settings, they remain there or moved to media or sd card.

_Media folders are not recommended by Android and we do not know how long they will be available._
Please use the default data dir as selection for the storage folder.

Cruiser 3.0.15 has again the internal data folder as default to avoid such problems. :slightly_smiling_face: 

---

Media folders are useful when we need to change the files, downloads do not need it.
And we do not know how OEM modify Android, so media may have extra restrictions.

**Comment by devemux86** [comment]
> particularly new Graphhopper files

GraphHopper graphs are zip files, they are downloaded in the internal storage and are extracted in the selected storage folder.

e.g. france.zip has a france folder inside, it is unzipped and then renamed to "france (Mapsforge)"
Interesting, so the rename did not work in the media folder of your device?

> Maybe some kind of "cache" within the Android OS?

Downloads are done by Android not the apps, we can watch them in the notification area with the :arrow_down: icons.
The apps just tell Android to start a download, wait for the download and then process the downloaded file.

**Comment by devemux86** [comment]
@MotoUKRider what happens with smaller GraphHopper graph files?
Are they extracted properly in the media folder?

**Reply by WolfgangZ42** [reply]
As far as I can tell, the download of graphs works fine:
<img width="400" src="https://user-images.githubusercontent.com/121806238/227803274-3a00c338-e775-4775-acae-7f6df1d47fdb.png"/>  
To be sure: I don't need this GraphHopper data at all if I have selected "BRouter (offline)" under settings/routing/routing - right?

**Comment by MotoUKRider** [comment]
Thanks for the info - I must have been confused so switched to "media" because I was under the impression *that* was the safe area for the file system going foward. Doh!

I cannot seem to consistently replicate the Graph download issue :(  I'll move my data back to "data" and try to break it again!

**Comment by 0709wiwiwi** [comment]


**Reply by 0709wiwiwi** [reply]
**A reference to use Android media as a "universal" best exchange solution between different gps apps.**
[Android media](https://forum.kurviger.de/t/offline-maps-are-not-integrated-and-old-tour-files-do-not-work-app1-14-xx/10532/10?u=0709)

Seems also Kurviger 1 Pro users by A11 and next need to be inventive to use already existing vector maps.  
Here the user "Biker" had already indicated a comfortable way to continue using the Locus vector maps. 
Provided you do read what he writes carefully, his solution does work. 

_The fiddling required to deal with modern Androids disappoints probably quite a few "normal" users imo. 
Multi app usage ? 
Gps app installs do not involve promises of eternal fidelity.
App developers who agree on the most elegant instant exchange methods would show respect for users  
As otherwise Garmin (and other) hardware gps salesmen are probably already rubbing their hands together._

**Comment by 0709wiwiwi** [comment]
Graph download issues in media #100

Or **Map and Poi and Graphs download issues in data and media** ?

> The download system on Graphs/Maps/POIs seems not to initiate a lot of the time

Probably is similar ?
https://github.com/devemux86/cruiser/discussions/33#discussioncomment-5426081

_I had noticed exactly that same experience too, but already enough people had reported this,  yes ?_ 

I (midnight) have now retested on the A10  (Xiamo Mi A2 Lite) phone.  
Need a test by a small Graph download ?  
Yes > Luxembourg > into **Android media** in the sd card.
Download is normal > installed > activated > Operation fine.
